crates/sqllib/src/casts.rs

```diff
@@ -2148,261 +2068,399 @@ pub fn negative(captures: &Captures) -> bool {
+        Ok(ShortInterval::new(minutes * 60 * 1000))
     } else {
-        panic!("Interval '{value}' does not have format 'days hours:minutes'");
+        Err(SqlRuntimeError::from_string(format!(
+            "Interval '{value}' does not have format 'days hours:minutes'",
+        )))
     }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_ShortInterval_HOURS_TO_MINUTES_s(value: String) -> ShortInterval {
+pub fn cast_to_ShortInterval_HOURS_TO_MINUTES_s(value: String) -> SqlResult<ShortInterval> {
     if let Some(captures) = HOURS_TO_MINUTES.captures(&value) {
         let negative = negative(&captures);
-        let hcap = captures.get(2).unwrap().as_str();
-        let hours: i64 = hcap
-            .parse()
-            .unwrap_or_else(|_| panic!("HOURS is not a number: {}", hcap));
+        let hourcap = captures.get(2).unwrap().as_str();
+        let hours = match hourcap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse HOURS: {}; {}",
+                    hourcap, e,
+                )))
+            }
+            Ok(hours) => hours,
+        };
         let mincap = captures.get(3).unwrap().as_str().trim();
-        let minutes = mincap
-            .parse()
-            .unwrap_or_else(|_| panic!("MINUTES is not a number: {}", mincap));
+        let minutes = match mincap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse MINUTES: {}; {}",
+                    mincap, e,
+                )))
+            }
+            Ok(minutes) => minutes,
+        };
         validate_minutes(minutes);
         let minutes = if negative {
             -hours * 60 - minutes
         } else {
             hours * 60 + minutes
         };
-        ShortInterval::new(minutes * 60 * 1000)
+        Ok(ShortInterval::new(minutes * 60 * 1000))
     } else {
-        panic!("Interval '{value}' does not have format 'hours:minutes'");
+        Err(SqlRuntimeError::from_string(format!(
+            "Interval '{value}' does not have format 'hours:minutes'",
+        )))
     }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_ShortInterval_SECONDS_s(value: String) -> ShortInterval {
+pub fn cast_to_ShortInterval_SECONDS_s(value: String) -> SqlResult<ShortInterval> {
     if let Some(captures) = SECONDS.captures(&value) {
         let negative = negative(&captures);
         let seccap = captures.get(2).unwrap().as_str().trim();
-        let seconds: i64 = seccap
-            .parse()
-            .unwrap_or_else(|_| panic!("SECONDS is not a number: {}", seccap));
+        let seconds = match seccap.parse::<i64>() {
+            Err(e) => {
+                return Err(SqlRuntimeError::from_string(format!(
+                    "Could not parse SECONDS: {}; {}",
+                    seccap, e,
+                )))
+            }
+            Ok(seconds) => seconds,
```

Comment:
sometimes you write the Err case first and sometimes you write the Ok case first, I wonder if there is a reason for it

Reply:
Tired to write so many stupid functions.